### PR TITLE
Lisa audio

### PR DESCRIPTION
**Pull request by libyrne**
Initial pull request to start code review. More pushes to come

### CODE REVIEW

File: vz_acoustic_scene_analysis/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+##   * add "message_runtime" and every package in MSG_DEP_SET to
+##     catkin_package(CATKIN_DEPENDS ...)
+##   * uncomment the add_*_files sections below as needed
+##     and list every .msg/.srv/.action file to be processed
+##   * uncomment the generate_messages entry below
+##   * add every package in MSG_DEP_SET to generate_messages(DEPENDENCIES ...)
+
+## Generate messages in the 'msg' folder
+add_message_files(
+  FILES
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/CMakeLists.txt:59]: Can we rename these to 'VZ_AudioInfo.msg' and 'VZ_AudioData.msg'?

File: vz_acoustic_scene_analysis/config/audio_params.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+TIMEOUT: 100000
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/config/audio_params.yaml:1]: Can we add a namespace here? Assume we may have many TIMEOUT parameters across different packages. 
So something like
VZ_ACCOUSTIC_SCENE_ANALYSIS:
   TIMEOUT:BLAH

File: vz_acoustic_scene_analysis/config/audio_params.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+VENDOR_ID: 0x2886
+PRODUCT_ID: 0x0018
+RESPEAKER_RATE: 16000
+RESPEAKER_CHANNELS: 6 # must flash 6_channels_firmware.bin first
+RESPEAKER_WIDTH: 2
+RESPEAKER_INDEX: get_respeaker_device_id()
+CHUNK: 1024
+chunk_size: 0.5
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/config/audio_params.yaml:9]: Keep them all CAPS?

File: vz_acoustic_scene_analysis/msg/MyAudioData.msg [hunk 4]
```diff
@@ -0,0 +1 @@
+uint16[] data
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/msg/MyAudioData.msg:1]: Sorry to be that way.. can u add a  comment as to what this data represents?

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py [hunk 5]
```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python3
+
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/scripts/stretch_audio.py:2]: Can you segregate the includes? As in ROS includes, python specific includes, custom includes etc? Just may look cleaner

File: vz_human_state_estimation/scripts/ros_interface.py [hunk 6]
```diff
@@ -34,7 +34,7 @@ def __init__(self):
         self.bridge = CvBridge()
 
         #Image subscribers
-        rgbd_sub    = message_filters.Subscriber("/D435i_raw",Image)
+        rgbd_sub    = message_filters.Subscriber("/D435i/image_raw",Image)
```
Comment (libyrne) [inline vz_human_state_estimation/scripts/ros_interface.py:37]: Load this as a parameter please?

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py [hunk 7]
```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python3
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/scripts/stretch_audio.py:1]: Can we call this file ros_interface.py? Expect to have 1 ros_interface.py for every new algorithm. See human state estimation.py file

File: vz_ros_wrappers/launch/D435i_and_Lepton.launch [hunk 8]
```diff
@@ -1,6 +1,8 @@
 <launch>
       <node pkg="vz_ros_wrappers" type="grab_D435i.py" name="D435i" />
       <node pkg="vz_ros_wrappers" type="grab_thermal.py" name="FLIRLepton" />
+      <node pkg="image_transport" type="republish" name="D435iCompressed"
```
Comment (libyrne) [inline vz_ros_wrappers/launch/D435i_and_Lepton.launch:4]: Just confirming, this is to take in the ros msg topic published by the default driver that comes with the robot?

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+            print('Respeaker not on USB bus')
+
+
+if __name__ == "__main__":
+    try:
+        audio = Audio()
+        audio.process_audio_loop()
+    except rospy.ROSInterruptException:
+        print('Audio processing node failed!')
+        pass
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/scripts/stretch_audio.py:289]: revoke the pass?

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+RESPEAKER_WIDTH = rospy.get_param("/RESPEAKER_WIDTH", 2)
+RESPEAKER_INDEX = get_respeaker_device_id()
+CHUNK = rospy.get_param("/CHUNK", 1024)
+
+class Audio:
+    def __init__(self):
+        # Initialiaze list to store audio segments
+        self.wav_list = []
+        self.record_count = 0 # Count how many times we've recorded f seconds of audio
+        self.file_name = "/home/hello-robot/Desktop/output_audio.wav"
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/scripts/stretch_audio.py:202]: Do we need this?

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+        p = pyaudio.PyAudio()
+
+        stream = p.open(rate=RESPEAKER_RATE,
+                        format=p.get_format_from_width(RESPEAKER_WIDTH),
+                        channels=RESPEAKER_CHANNELS,
+                        input=True,
+                        input_device_index=RESPEAKER_INDEX,
+                        output= False)
+
+        frames = []
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/scripts/stretch_audio.py:236]: Can we split this calculation to a new variable to make it clear what this represents? int(RESPEAKER_RATE / CHUNK * seconds)

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+        for i in range(0, int(RESPEAKER_RATE / CHUNK * seconds)):
+            data = stream.read(CHUNK)
+            a = np.frombuffer(data,dtype=np.uint16)[0::6] # extracts fused channel 0
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/scripts/stretch_audio.py:239]: Curious wat 0::6 represents

File: vz_acoustic_scene_analysis/config/audio_params.yaml (hunk 2 above)
Comment (libyrne) [reply vz_acoustic_scene_analysis/config/audio_params.yaml:1]: Also add units to the name of the parameters

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py [hunk 13]
```diff
@@ -0,0 +1,289 @@
...
+        # audio = MyAudioData()
+        # audio.data = np.array(recorded_frames, dtype=np.uint16)
+        # self.audio_data_pub.publish(audio)
+        self.wav_list.append(recorded_frames)
+        self.record_count += 1
+        # Every 5 seconds for 
+        if ((self.record_count % (self.secs/self.chunk_size)) == 0): # set param sequence size
+            return_list =  self.wav_list
+            # Remove 0.2 seconds of audio data
+            self.wav_list.pop(0)
```
Comment (libyrne) [inline vz_acoustic_scene_analysis/scripts/stretch_audio.py:219]: Can u write a comment explaining the 2 pops?

**Review by libyrne — COMMENTED** [review]
initial code comments

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py (hunk 10 above)
Comment (libyrne) [reply vz_acoustic_scene_analysis/scripts/stretch_audio.py:202]: @trivediana which one are you asking about? I removed the file_name variable

File: vz_ros_wrappers/launch/D435i_and_Lepton.launch [hunk 14]
```diff
@@ -1,6 +1,8 @@
...
+            args="raw in:=/D435i/image_raw compressed out:=/D435i/image_raw" />
       <!-- Flip images and display when collecting data? -->
       <param name="flip_images_grab_data" type="bool" value="true" />
```
[the remark this answers is not in the document]
Comment (libyrne) [reply vz_ros_wrappers/launch/D435i_and_Lepton.launch:7]: @trivediana just a reminder of this comment

File: vz_acoustic_scene_analysis/scripts/stretch_audio.py (hunk 12 above)
Comment (libyrne) [reply vz_acoustic_scene_analysis/scripts/stretch_audio.py]: @trivediana  [0::6] takes every sixth object from the array

Comment (trivediana) [reply vz_acoustic_scene_analysis/scripts/stretch_audio.py]: I see. 
Can we make this not hardcoded? Something like 
`var = something that gives 6`
 a = np.frombuffer(data,dtype=np.uint16)[0::var] # extracts fused channel 0

**Pull request closed without merging**